src/ChanperfTracker.ts

```diff
@@ -0,0 +1,145 @@
+        let dataChanged = false;
+        let lines = logLine.split('\n');
+        let normalOutput = '';
+        let chanperfHistory: ChanperfEventData;
+
+        for (let line of lines) {
+            let infoAvailableMatch = /channel:\smem=([0-9]+)kib{[a-z]+=([0-9]+),[a-z]+=([0-9]+),[a-z]+=([0-9]+)},\%cpu=([0-9]+){[a-z]+=([0-9]+),[a-z]+=([0-9]+)}/gmi.exec(line);
+            // see the following for an explanation for this regex: https://regex101.com/r/AuQOxY/1
+            if (infoAvailableMatch) {
+                let [fullMatch, totalMemKib, anonMemKib, fileMemKib, sharedMemKib, totalCpuUsage, userCpuUsage, sysCpuUsage] = infoAvailableMatch;
+                chanperfHistory = this.createNewChanperfHistory();
+                chanperfHistory.error = null;
+
+                chanperfHistory.memory = {
+                    total: parseInt(totalMemKib) * 1024,
+                    anonymous: parseInt(anonMemKib) * 1024,
+                    file: parseInt(fileMemKib) * 1024,
+                    shared: parseInt(sharedMemKib) * 1024
+                };
+
+                chanperfHistory.cpu = {
+                    total: Math.min(parseInt(totalCpuUsage), 100),
+                    user: Math.min(parseInt(userCpuUsage), 100),
+                    system: Math.min(parseInt(sysCpuUsage), 100)
+                };
+
+                if (!this.filterOutLogs) {
+                    normalOutput += line + '\n';
+                }
+                this.emit('chanperf-event', chanperfHistory);
+                this.chanperfHistory.push(chanperfHistory);
+            } else {
+                // see the following for an explanation for this regex: https://regex101.com/r/Nwqd5e/1/
+                let noInfoAvailableMatch = /channel:\s(mem\sand\scpu\sdata\snot\savailable)/gim.exec(line);
+
+                if (noInfoAvailableMatch) {
+                    chanperfHistory = this.createNewChanperfHistory();
+                    chanperfHistory.error = { message: noInfoAvailableMatch[1] };
+
+                    if (!this.filterOutLogs) {
+                        normalOutput += line + '\n';
+                    }
+                    this.emit('chanperf-event', chanperfHistory);
+                    this.chanperfHistory.push(chanperfHistory);
+                } else if (line) {
+                    normalOutput += line + '\n';
+                }
+            }
+
+        }
+
+        return normalOutput;
+    }
+
+    /**
+     * Helper function used to create a new ChanperfEventData object with default values
+     */
+    private createNewChanperfHistory(): ChanperfEventData {
+        return {
+            memory: {
+                total: 0,
+                anonymous: 0,
+                file: 0,
+                shared: 0
+            },
+            cpu: {
+                total: 0,
+                user: 0,
+                system: 0
+            }
+        };
+    }
+}
+
+export interface ChanperfEventData {
```

Comment:
What about just `ChanperfData` or `ChanperfDatum` ("datum" is singular word for plural word "data") or even just call it `Chanperf`. (i.e. "its a struct, it's a class, it's a chanperf").